extensions/tls-registry/runtime/src/main/java/io/quarkus/tls/runtime/TlsCertificateUpdater.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.tls.runtime;
+
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import jakarta.enterprise.event.Event;
+import jakarta.enterprise.inject.spi.CDI;
+
+import io.quarkus.tls.CertificateUpdatedEvent;
+import io.quarkus.tls.TlsConfiguration;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+/**
+ * A helper class that reload the TLS certificates at a configured interval.
+ * When the certificate is reloaded, a {@link CertificateUpdatedEvent} is fired.
+ */
+public class TlsCertificateUpdater {
+
+    private Vertx vertx;
+    private CopyOnWriteArrayList<Long> tasks;
+    private Event<CertificateUpdatedEvent> event;
```

Comment:
Do these need to be `volatile`?

Reply:
No, `init` won't be called concurrently to any other call. We could even have them final (if `init` becomes the constructor)